packages/sdk/src/engine-commands/getConfig.ts

```diff
@@ -34,128 +37,278 @@ export type GetConfigOptions = {
+           * to serialize big datamodels.
+           */
+          PRISMA_DML_PATH: tempDatamodelPath,
+          RUST_BACKTRACE: '1',
+        },
+        maxBuffer: MAX_BUFFER,
       }
-    }
-    const engineArgs = []
-
-    const args = options.ignoreEnvVarErrors ? ['--ignoreEnvVarErrors'] : []
-
-    const result = await execa(queryEnginePath, [...engineArgs, 'cli', 'get-config', ...args], {
-      cwd: options.cwd,
-      env: {
-        PRISMA_DML_PATH: options.datamodelPath ?? tempDatamodelPath,
-        RUST_BACKTRACE: '1',
-      },
-      maxBuffer: MAX_BUFFER,
-    })
 
-    if (tempDatamodelPath) {
-      await unlink(tempDatamodelPath)
-    }
-
-    data = JSON.parse(result.stdout)
-  } catch (e: any) {
-    if (e.stderr || e.stdout) {
-      const error = e.stderr ? e.stderr : e.stout
-      let jsonError, message
-      try {
-        jsonError = JSON.parse(error)
-        message = `${chalk.redBright(jsonError.message)}\n`
-        if (jsonError.error_code) {
-          if (jsonError.error_code === 'P1012') {
-            message = chalk.redBright(`Schema Parsing ${jsonError.error_code}\n\n`) + message
-          } else {
-            message = chalk.redBright(`${jsonError.error_code}\n\n`) + message
+      const engineArgs = []
+      const args = options.ignoreEnvVarErrors ? ['--ignoreEnvVarErrors'] : []
+
+      return TE.tryCatch(
+        () => {
+          if (process.env.FORCE_PANIC_QUERY_ENGINE_GET_CONFIG) {
+            debug('Triggering a Rust panic...')
+            return execa(
+              queryEnginePath,
+              [...engineArgs, 'cli', 'debug-panic', '--message', 'FORCE_PANIC_QUERY_ENGINE_GET_CONFIG'],
+              execaOptions,
+            )
           }
-        }
-      } catch (e) {
-        // if JSON parse / pretty handling fails, fallback to simple printing
-        throw new GetConfigError(error)
-      }
 
-      throw new GetConfigError(message)
-    }
+          return execa(queryEnginePath, [...engineArgs, 'cli', 'get-config', ...args], execaOptions)
+        },
+        (e) => ({
+          type: 'execa' as const,
+          reason: 'Error while interacting with query-engine binary',
+          error: e as execa.ExecaError,
+        }),
+      )
+    })(),
+    TE.map((result) => ({ result })),
+    TE.chainW(({ result }) => {
+      return pipe(
+        E.tryCatch(
+          () => JSON.parse(result.stdout) as ConfigMetaFormat,
+          (e) => ({
+            type: 'parse-json' as const,
+            reason: 'Unable to parse JSON',
+            error: e as Error,
+          }),
+        ),
+        TE.fromEither,
+      )
+    }),
+  )
 
-    throw new GetConfigError(e)
+  const configEither = await pipeline()
+
+  if (E.isRight(configEither)) {
+    debug('config data retrieved without errors in getConfigBinary')
+    await unlinkTempDatamodelPath(options, tempDatamodelPath)()
```

Comment:
```suggestion
    // Unlink the file only if no error occured since sendPanic will need it later
    await unlinkTempDatamodelPath(options, tempDatamodelPath)()
```